virtualmachine/vsphere/util.go

```diff
@@ -603,7 +669,22 @@ var createTemplateName = func(t string, ds string) string {
 var uploadTemplate = func(vm *VM, dcMo *mo.Datacenter, selectedDatastore string) error {
 	template := createTemplateName(vm.Template, selectedDatastore)
 	vm.datastore = selectedDatastore
+	basePath, err := ioutil.TempDir("", "")
+	if err != nil {
+		return err
+	}
+	defer func() {
+                if err := os.RemoveAll(basePath); err != nil {
+                        fmt.Errorf("can't remove temp directory, error: %s", err.Error())
+                }
+        }()
 	// Read the ovf file
```

Comment:
The intention is to introduce ova file without disturbing the current working with OvfPath. 
If OvaPathUrl is given then OvaPathUrl will be used, if not then OvfPath will be used. If Both are given preference will be given to OvaPathUrl.
OvfPath - refers to locally available ovf file
OvaPathUrl - can be remote or local ova file